src/utils/parse-condition.ts

```
@@ -5,7 +5,7 @@ import type {
 } from '@tazama-lf/frms-coe-lib/lib/interfaces/event-flow/ConditionDetails';
 import type { RawConditionResponse } from '@tazama-lf/frms-coe-lib/lib/interfaces/event-flow/EntityConditionEdge';
 
-export const parseConditionEntity = (input: RawConditionResponse[]): EntityConditionResponse => {
+export const parseConditionEntity = (input: RawConditionResponse[], tenantId = 'DEFAULT'): EntityConditionResponse => {
```

I see most of the functions in this file allow for the `tenantId` to be defaulted to the DEFAULT `tenantId` in the function call (this also means that the parameter is now technically considered optional). This seems like a risk, since if someone were to call this function without a valid `tenantId` in the code, the system will assume that it should make it the default value, but at the point where this function is called, the system should already have resolved the `tenantId` - either from the Tazama JWT, or by adding the default earlier (and once-off) in the code at the same point where it evaluates the environment variable for authentication (which I expect would be around the same time as it would try to resolve it from the Tazama JWT in the new tenantId middleware function). When this function is called, if no `tenantId` is provided, the system should either throw an error (exception-handling) or fail to build the package entirely (build error).

---

You’re right—`tenantId` should always be required, never defaulted. Thanks for catching this. I will fix any remaining defaults to ensure strict tenant isolation.